src/service/resources-impl.js

```diff
@@ -1327,6 +1327,13 @@ export class Resources {
     // Breadth-first search.
     if (element.classList.contains('-amp-element')) {
       callback(this.getResourceForElement(element));
+      // Also schedule amp-element that is a placeholder for the element.
+      if (!element.classList.contains('-amp-layout')) {
```

Comment:
I think it was just to avoid discovering it multiple times as you mentioned, but explanation sounds good to me! Dropped it.